toolchain/check/handle_class.cpp

```diff
@@ -166,6 +167,98 @@ auto HandleClassDefinitionStart(Context& context, Parse::NodeId parse_node)
   return true;
 }
 
+auto HandleBaseIntroducer(Context& /*context*/, Parse::NodeId /*parse_node*/)
+    -> bool {
+  return true;
+}
+
+auto HandleBaseColon(Context& /*context*/, Parse::NodeId /*parse_node*/)
+    -> bool {
+  return true;
+}
+
+auto HandleBaseDecl(Context& context, Parse::NodeId parse_node) -> bool {
+  auto base_type_expr_id = context.node_stack().PopExpr();
+
+  auto class_id =
+      context.node_stack().PeekIf<Parse::NodeKind::ClassDefinitionStart>();
+  if (!class_id) {
+    CARBON_DIAGNOSTIC(BaseOutsideClass, Error,
+                      "`base` declaration can only be used in a class.");
+    context.emitter().Emit(parse_node, BaseOutsideClass);
+    return true;
+  }
+
+  auto& class_info = context.classes().Get(*class_id);
+  if (class_info.base_id.is_valid()) {
+    CARBON_DIAGNOSTIC(
+        BaseRepeated, Error,
+        "Repeated `base` declaration. Multiple inheritance is not permitted.");
```

Comment:
"Repeated" to me implies duplicative, including the base class name. Maybe "Multiple `base` declarations found"?

Though you could also say something like "Cannot have multiple `base` declarations." or "Only one `base` declaration allowed."

```suggestion
        "Multiple `base` declarations found. Multiple inheritance is not permitted.");
```

Reply:
Done.